### PR TITLE
Make changes for jetpack datastore broker support

### PR DESCRIPTION
**Pull request by rpdome**
Related: https://github.com/AzureAD/ad-accounts-for-android/pull/2197

### CODE REVIEW

File: config/spotbugs/exclude.xml [hunk 1]
```diff
@@ -32,4 +32,9 @@
     <Package name="com.microsoft.identity.common.java.commands.*" />
     <Bug pattern="RV_RETURN_VALUE_IGNORED_NO_SIDE_EFFECT" />
 </Match>
+    <Match>
+        <!-- Spotbugs doesn't seem to work well with kotlin...
+        https://github.com/spotbugs/spotbugs/issues/573 -->
+        <Source name="~.*\.kt"/>
+    </Match>
```
Comment (fadidurah) [inline config/spotbugs/exclude.xml:39]: I'm confused what this addition does

File: common4j/src/main/com/microsoft/identity/common/java/crypto/IKeyAccessorStringAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.java.crypto
+
+import com.microsoft.identity.common.java.exception.ClientException
+
+interface IKeyAccessorStringAdapter {
```
Comment (fadidurah) [inline common4j/src/main/com/microsoft/identity/common/java/crypto/IKeyAccessorStringAdapter.kt:27]: Should we set this as an interface for `KeyAccessorStringAdapter`

**Review by fadidurah — APPROVED** [review]
LGTM

File: config/spotbugs/exclude.xml (hunk 1 above)
Comment (fadidurah) [reply config/spotbugs/exclude.xml:39]: nevermind, we just ignore the .kt classes

File: common4j/src/main/com/microsoft/identity/common/java/crypto/IKeyAccessorStringAdapter.kt (hunk 2 above)
Comment (rpdome) [reply common4j/src/main/com/microsoft/identity/common/java/crypto/IKeyAccessorStringAdapter.kt:27]: Yes. I renamed the original class to UTF8KeyAccessorStringAdapter

**Review by melissaahn — APPROVED** [review]

**Pull request merged**